src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -101,16 +101,57 @@ internal CloudTable GetDiagnosticEventsTable(DateTime? now = null)
             return _diagnosticEventsTable;
         }
 
-        private static string GetCurrentTableName(DateTime now)
+        private static string GetTableName(DateTime date)
         {
-            return $"{TableNamePrefix}{now:yyyyMM}";
+            return $"{TableNamePrefix}{date:yyyyMM}";
         }
 
         protected internal virtual async void OnFlushLogs(object state)
         {
             await FlushLogs();
         }
 
+        // TODO:// Call this ~1 minute after host startup to avoid blocking host start & impacting cold start.
+        private async void PurgePreviousEventVersions()
+        {
+            _logger.LogInformation("Purging previous diagnostic event versions.");
+
+            try
+            {
+                var tables = await TableStorageHelpers.ListTablesAsync(TableClient, TableNamePrefix);
+
+                TableQuery<DiagnosticEvent> eventVersionExistsQuery = new TableQuery<DiagnosticEvent>()
+                    .Where(TableQuery.GenerateFilterCondition(nameof(DiagnosticEvent.EventVersion), QueryComparisons.NotEqual, string.Empty));
+
+                TableQuery<DiagnosticEvent> eventVersionOutdatedQuery = new TableQuery<DiagnosticEvent>()
+                    .Where(TableQuery.GenerateFilterCondition(nameof(DiagnosticEvent.EventVersion), QueryComparisons.LessThan, DiagnosticEvent.CurrentEventVersion));
+
+                foreach (var table in tables)
+                {
+                    var eventVersionExists = await table.ExecuteQuerySegmentedAsync(eventVersionExistsQuery, null);
+
+                    // Delete table if it doesn't have records with event version
+                    if (eventVersionExists.Results.Count == 0)
```

Comment:
Could this check lead to scenarios where different instances will be deleting a table that was just created to persist new entries (with the new version)? If the table is empty, it could be that it was about to be used by a different instance to persist data.

Reply:
This is a great point, yes it would delete those too I believe. There is no way to check for if a "column" exists though :(

Reply:
We could just delete the records here instead of the entire table? But that might be excessive

Reply:
Okay I've refactored this to ignore tables that have 0 records - thoughts?